lib/models/package-info-cache/package-info.js

```diff
@@ -5,6 +5,24 @@ const ErrorList = require('./error-list');
 const Errors = require('./errors');
 const AddonInfo = require('../addon-info');
 
+function lexicographically(a, b) {
+  if (typeof a.name === 'string' && typeof b.name === 'string') {
+    return a.name.localeCompare(b.name);
+  }
+  return 0;
+}
+
+// lastWriteWins
```

Comment:
unit test

Reply:
a.name == "foo" and b.name == undefined is 0? that seems weird.

Reply:
Do we expect this list to contain undefined?

Reply:
> a.name == "foo" and b.name == undefined is 0? that seems weird.

Ya, I agree.

> Do we expect this list to contain undefined?

It does for semi annoying reasons. I didn't really want to fix the root cause, but I think I will now. As that makes this feel less shitty and fixes @krisselden's issue